shopinvader_image/models/shopinvader_image_mixin.py

```diff
@@ -100,18 +104,33 @@ def _get_image_url_key(self, image_relation):
     def _get_image_data_for_record(self):
         self.ensure_one()
         res = []
+        failed = {}
         resizes = self._resize_scales()
         for image_relation in self[self._image_field]:
             url_key = self._get_image_url_key(image_relation)
             image_data = {}
             for resize in resizes:
-                thumbnail = image_relation.image_id.get_or_create_thumbnail(
-                    resize.size_x, resize.size_y, url_key=url_key
-                )
-                image_data[resize.key] = self._prepare_data_resize(
-                    thumbnail, image_relation
-                )
+                try:
+                    thumbnail = image_relation.image_id.get_or_create_thumbnail(
+                        resize.size_x, resize.size_y, url_key=url_key
+                    )
+                except FileNotFoundError:
+                    # If get/set the file to/from external storage fails
+                    # skip the image and do not let the whole data compute fail
+                    failed.setdefault(tuple(self.ids), []).append(
+                        f"{resize.size_x}x{resize.size_y}"
+                    )
+                else:
+                    image_data[resize.key] = self._prepare_data_resize(
+                        thumbnail, image_relation
+                    )
             res.append(image_data)
+        if failed:
+            _logger.error(
```

Comment:
IMO we can't silently ignore errors in production. Being able to ignore this kind of error when you are on your dev env or some non critical envs could ease your live but by default it must not be possible to publish new items to the website if images are broken. -> This behavior should be managed by configuration (backend)

Reply:
@lmignon this is why I set the PR in draft and I'm looking for feedback :wink: 
IMO we should have something similar to PIM: product info completeness is computed and you don't export if is not complete. @Cedric-Pigeon have you ever did/thought something about this?

Regarding images: yes, they are important but they are not critical all the times.
Of course, if you have images that are tied to specific variants they could be misleading.
Still, I find prices and other info more relevant -> if you break the sync because an image is not available it worse IMO.
It also depends on the state: is the product new? Is already on the shop?

Anyway, for now I'll add a way to ignore missing records via on storage_backend, only for dev purpose.
Then we can discuss about improvements in this area at the sprint in June maybe.